File: for-raspberry-pi/python/14_AnswerMachine.py
#!/usr/bin/env python3

import RPi.GPIO as GPIO
import time

BeepPin = 17
ResetBtnPin = 18
BtnPin =(27,22,23)
LedPin =(5,6,13,19)


def print_message():
    print ("========================================")
    print ("|            Answer Machine            |")
    print ("|    ------------------------------    |")
    print ("|        Buzzer connect to BCM 17      |")
    print ("|  Buttons connect to BCM 18 27 22 23  |")
    print ("|    LEDs connect to BCM 5 6 13 19     |")
    print ("|                                      |")
    print ("|         Make a Answer Machine        |")
    print ("|                                      |")
    print ("|                            SunFounder|")
    print ("========================================\n")
    print ('Program is running...')
    print ('Please press Ctrl+C to end the program...')
    input ("Press Enter to begin\n")

def setup():
    GPIO.setmode(GPIO.BCM)
    GPIO.setup(BeepPin, GPIO.OUT, initial=GPIO.LOW)
    GPIO.setup(ResetBtnPin, GPIO.IN)
    GPIO.setup(LedPin[0], GPIO.OUT, initial=GPIO.HIGH)
    for i in range(1,4):
        GPIO.setup(LedPin[i], GPIO.OUT, initial=GPIO.LOW)
    for i in range(0,3):
        GPIO.setup(BtnPin[i], GPIO.IN)

def Alarm():
    for i in range(0,50):
        GPIO.output(BeepPin,GPIO.HIGH)
        time.sleep(0.003)
        GPIO.output(BeepPin,GPIO.LOW)
        time.sleep(0.003)


def loop():
    print_message()
    flag = 1
    while True:
        if GPIO.input(ResetBtnPin) == 1:
            flag = 1
            GPIO.output(LedPin[0],GPIO.HIGH)
            GPIO.output(LedPin[1],GPIO.LOW)
            GPIO.output(LedPin[2],GPIO.LOW)
            GPIO.output(LedPin[3],GPIO.LOW)
        if flag == 1:
            if GPIO.input(BtnPin[0]) == 1:
                flag = 0
                GPIO.output(LedPin[0],GPIO.LOW)
                Alarm()
                GPIO.output(LedPin[1],GPIO.HIGH)
                GPIO.output(LedPin[2],GPIO.LOW)
                GPIO.output(LedPin[3],GPIO.LOW)
            elif GPIO.input(BtnPin[1]) == 1:
                flag = 0
                GPIO.output(LedPin[0],GPIO.LOW)
                Alarm()
                GPIO.output(LedPin[1],GPIO.LOW)
                GPIO.output(LedPin[2],GPIO.HIGH)
                GPIO.output(LedPin[3],GPIO.LOW)
            elif GPIO.input(BtnPin[2]) == 1:
                flag = 0
                GPIO.output(LedPin[0],GPIO.LOW)
                Alarm()
                GPIO.output(LedPin[1],GPIO.LOW)
                GPIO.output(LedPin[2],GPIO.LOW)
                GPIO.output(LedPin[3],GPIO.HIGH)
            



def destroy():
    # Turn off buzzer
    GPIO.output(BeepPin, GPIO.LOW)
    GPIO.output(LedPin[0],GPIO.LOW)
    GPIO.output(LedPin[1],GPIO.LOW)
    GPIO.output(LedPin[2],GPIO.LOW)
    GPIO.output(LedPin[3],GPIO.HIGH)
    # Release resource
    GPIO.cleanup()    

# If run this script directly, do:
if __name__ == '__main__':
    setup()
    try:
        loop()
    # When 'Ctrl+C' is pressed, the child program 
    # destroy() will be  executed.
    except KeyboardInterrupt:
        destroy()

File: for-raspberry-pi/c/Lesson_14_AnswerMachine/14_AnswerMachine.c
/**********************************************************************
* Filename    : 14_AnswerMachine.c
* Description : Make a AnswerMachine.
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define BeepPin 0
#define ResetBtnPin 1
const int BtnPin[] = {2,3,4};
const int LedPin[] = {21,22,23,24};

void Alarm()
{
    for(int i=0;i<50;i++){
    digitalWrite(BeepPin,HIGH); //the buzzer sound
    delay(2); //delay 2ms
    digitalWrite(BeepPin,LOW);  //without sound
    delay(2);       //when delay time changed,the frequency changed
  }
}

int main(void){
	if(wiringPiSetup() == -1){ //when initialize wiring failed, print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}
	
	 pinMode(BeepPin, OUTPUT);   //set GPIO0 output
	for(int j=1;j<4;j++)
    {
        pinMode(LedPin[j], OUTPUT);
        digitalWrite(LedPin[j],LOW);
    }
    pinMode(LedPin[0], OUTPUT);
    digitalWrite(LedPin[0],HIGH);
    for(int k;k<3;k++)
    {
        pinMode(BtnPin[k], INPUT);
    }

	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|            Answer Machine            |\n");
	printf("|    ------------------------------    |\n");
	printf("|        Buzzer connect to GPIO 0      |\n");
	printf("|    Buttons connect to GPIO 1 to 4    |\n");
    printf("|    LEDs connect to GPIO 21 to 24     |\n");
	printf("|                                      |\n");
	printf("|         Make a Answer Machine        |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");

    int flag = 1;
    
	while(1){
        // if reset button is pressed
        if(digitalRead(ResetBtnPin) == 1)
        {
                flag = 1;              
                digitalWrite(LedPin[0], HIGH);//Reset Led turns on
                digitalWrite(LedPin[1],LOW);
                digitalWrite(LedPin[2],LOW);
                digitalWrite(LedPin[3],LOW);           
        }
        if(flag==1)
        {       
            //If the button1 press the first
            if(digitalRead(BtnPin[0]) == 1)
            {
                    flag = 0;
                    digitalWrite(LedPin[0],LOW);
                    Alarm();  //buzzer sound
                    digitalWrite(LedPin[1],HIGH);//turn the LED1 on only
                    digitalWrite(LedPin[2],LOW);
                    digitalWrite(LedPin[3],LOW);
                    while(digitalRead(ResetBtnPin));//detect the resetbtn,if pressed,out of the while loop           
            }
             if(digitalRead(BtnPin[1]) == 1)
            {        
                    flag = 0;
                    digitalWrite(LedPin[0],LOW);
                    Alarm();  //buzzer sound
                    digitalWrite(LedPin[1],LOW);
                    digitalWrite(LedPin[2],HIGH);//turn the LED2 on only
                    digitalWrite(LedPin[3],LOW);
                    while(digitalRead(ResetBtnPin));             
            }
             if(digitalRead(BtnPin[2]) == 1)
            {                              
                    flag = 0;
                    digitalWrite(LedPin[0],LOW);
                    Alarm();  //buzzer sound
                    digitalWrite(LedPin[1],LOW);
                    digitalWrite(LedPin[2],LOW);
                    digitalWrite(LedPin[3],HIGH);//turn the LED3 on only
                    while(digitalRead(ResetBtnPin));                
            }          
        }		
	}
	return 0;
}

